### ISSUE TITLE
UTC Time Bug

### ISSUE THREAD

**Issue by martinheppner** [bahnzumberg/zuugle-suchseite#419]
Look a the screenshot. The yellow marked times should be the same,  it they differ by two hours.

![Screenshot_20240813_093213_Gmail~2.jpg](https://github.com/user-attachments/assets/86d89643-e84d-45cd-924f-bd3424570e9f)

I received this screenshot by someone else, but I can see the bug myself at https://www.zuugle.at/tour/462/wien as well.

**Comment by Falsal** [comment]
The source for the "07:02" example above comes from the column : " connection_departure_datetime " while the column "connection_description_json" creates the actual itinerary inside the dropdown. So assuming that the latter is the correct one , i looked for other column names as an alternative source for the time that matches what the  "connection_description_json" provides and found column "connection_departure_datetime_entry" with the following example:  "2024-08-13T08:46:00+02:00". so i replaced the old column "connection_departure_datetime" with this new one.
Note that the +02:00 means it is 2 Hrs ahead of UTC (offset).



![Image](https://github.com/user-attachments/assets/37cc0b3b-b7cf-45aa-a7ee-4c8d0f76d069)

**Comment by martinheppner** [comment]
Now the traveltime there is ok, but travel time back is now wrong

![image](https://github.com/user-attachments/assets/b94b7914-4d82-4c14-9932-a027e3076569)

**Comment by Falsal** [comment]
are you sure that  the return time was matching  ?
Anyway, here are the data from the connection in console log, using my example above. 
This time theere is no column that gives the time matching the json data.


![Image](https://github.com/user-attachments/assets/3d1e0e62-55af-4988-bf92-0f611a21f775)

**Comment by martinheppner** [comment]
In tours.js, in function connectionsExtendedWrapper I switched the knex query to a knex.raw query and changed the timezone then to Europe/Berlin. On my local development this is working. After I deploayed to UAT, I had to realease, that it is not running on UAT. Do not know why - I am too tired to continue.

```
let connections = [];
    const fahrplan_sql = `SELECT * FROM fahrplan WHERE hashed_url='${tour.hashed_url}' AND city_slug='${city}' ORDER BY return_row ASC;`;
    const fahrplan_result = await knex.raw(fahrplan_sql)    
    if (!!fahrplan_result && !!fahrplan_result.rows) {
        connections = fahrplan_result.rows.map(connection => {
            connection.return_departure_datetime = momenttz(connection.return_departure_datetime).tz('Europe/Berlin').format();
            return connection;
        });
    }
```

**Comment by martinheppner** [comment]
Error from yesterday is mysterious: GitHub Actions ignored the new module while deploying (propably tired, too). So the backend crashed, as the module was missing. Today I realized that, by looking at the logs on the server. I added an empty line, so it would be recognized as a change of package.json and redeployed. Worked right away. Did the same now withe the connection_departure_datetime and removed connection_departure_datetime_entry.

Works. Already deployed to PROD.
